pages/_app.js

```diff
@@ -79,64 +59,41 @@ export default class Karbon14 extends App {
-                            <Signalhub.Consumer>
-                              {({ messages, channel, broadcast }) => (
-                                <EthereumProvider contracts={getProofLifeContract(env.NETWORK)}>
-                                  {({ accounts = {}, deployedContracts = {}, web3 }) => (
-                                    <ReactionComponent
-                                      initialState={{
-                                        scribes: []
-                                      }}
-                                      deployedContracts={deployedContracts}
-                                      didUpdate={({ props, prevProps, setState }) => {
-                                        if (!isEqual(props.deployedContracts, prevProps.deployedContracts))
-                                          updateUI({ deployedContracts, accounts, setState, web3 })
-                                      }}
-                                      render={({ state }) => {
-                                        const routerNext = { currentRoute }
-                                        const language = { getTranslation, selectedLanguage }
-                                        const signalHub = { messages, channel, broadcast }
-                                        const ethereum = { accounts, deployedContracts, web3 }
-                                        const ipfs = { addData }
-                                        const proofLifeContract = { ...state }
 
-                                        const currentProps = {
-                                          routerNext,
-                                          language,
-                                          signalHub,
-                                          ethereum,
-                                          proofLifeContract,
-                                          ipfs,
-                                          ...pageProps
-                                        }
-                                        return <Component {...currentProps} />
-                                      }}
-                                    />
-                                  )}
-                                </EthereumProvider>
-                              )}
-                            </Signalhub.Consumer>
-                          )}
-                        </LanguageContext.Consumer>
-                      </LanguageProvider>
-                    )}
-                  </RouterNext.Consumer>
-                )}
-              </Ipfs.Consumer>
-              <ToastContainer theme={theme} hideProgressBar />
-            </Ipfs.Provider>
-          </Signalhub.Provider>
-        </RouterNext.Provider>
+        <Providers contract={getProofLifeContract(env.NETWORK)} pathname={pathname}>
+          {({ translations, ethereum, routerNext, signalHub, ipfs }) => {
+            const { accounts = {}, deployedContracts = {}, web3 } = ethereum
```

Comment:
With all the Providers merged, now code looks much more clear, and only on component <Providers /> can expose them all

(Also refactored all the providers to avoid Consumer usage view/container side here)